chain/chain/src/runtime/tests.rs

```diff
@@ -1601,6 +1603,11 @@ fn prepare_transactions(
 /// Check that transactions validation works the same when using recorded storage proof instead of db.
 #[test]
 fn test_prepare_transactions_storage_proof() {
+    if !checked_feature!("stable", StatelessValidationV0, PROTOCOL_VERSION) {
+        println!("Test not applicable without StatelessValidation enabled");
```

Comment:
Does this mean that this test won't kick in before hitting testnet (with stable build)? isn't it too late?

Reply:
This means that this test will only execute in builds where the default protocol version has stateless validation enabled.
In practice this means that the test will be enabled for builds with `--features statelessnet_protocol` and `--features nightly`. Both of those options are present in the CI, so the test will be run on every change, and it will be enabled on statelessnet builds.

Tbh I don't know what `"stable"` in `checked_feature` is supposed to mean, the macro just checks the protocol version:
```rust
#[macro_export]
macro_rules! checked_feature {
    ("stable", $feature:ident, $current_protocol_version:expr) => {{
        $crate::version::ProtocolFeature::$feature.protocol_version() <= $current_protocol_version
    }};
```

The code is here if anyone wants to take a look: https://github.com/near/nearcore/blob/f2e87c0c1a4cd6484ce6003fdbfbdb8307221d07/core/primitives-core/src/version.rs#L224-L256

Reply:
We already have this kind of checks in many other tests that don't work without stateless validation, e.g: https://github.com/near/nearcore/blob/f2e87c0c1a4cd6484ce6003fdbfbdb8307221d07/integration-tests/src/tests/client/features/stateless_validation.rs#L33